src/foremast/utils/tasks.py

```diff
@@ -99,3 +98,48 @@ def check_task(taskid):
         raise SpinnakerTaskError(task_state)
     else:
         raise ValueError
+
+
+def check_task(taskid, timeout=DEFAULT_TASK_TIMEOUT):
+    """wrapper for check_task
+
+    Args:
+        taskid (str): the task id returned from post_task 
+        timeout (int) (optional): how long to wait before failing the task
+
+    Returns:
+        polls for task status.
+    """
+    wait = 2
+    max_attempts = int(timeout / wait)
+    return retry_call(partial(_check_task, taskid),
+                      max_attempts=max_attempts,
+                      wait=wait,
+                      exceptions=(AssertionError, ValueError))
+
+def wait_for_task(task_data):
```

Comment:
I do like that this simplifies the existing model of `id = post_task(); check_task(id)`.